autoprecompiles/src/lib.rs

```diff
@@ -560,14 +546,43 @@ pub fn optimize_precompile<T: FieldElement>(mut machine: SymbolicMachine<T>) ->
                 Some(addr) => addr,
             };
 
-            match mem_int.op {
-                MemoryOp::Read => {}
-                MemoryOp::Write => {
-                    last_store.insert(addr, i);
+            if receive {
+                match local_reg_mem.get(&addr) {
+                    Some(data) => {
```

Comment:
This block was unmodified, a loop below was removed but github diff is confused, as usual